### PR TITLE
Add Alert reference field

### PR DESCRIPTION
**Pull request by psiinon**
Add new alertRef field to alerts.
This is to allow us to tell the difference between different alerts raised by the same rule. By default is will be the plugin id - rules will need to be changed to append a qualifier for each of the alerts they raise.

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

File: zap/src/main/java/org/parosproxy/paros/core/scanner/Alert.java [hunk 1]
```diff
@@ -921,6 +942,49 @@ public void setSource(Source source) {
         this.source = source;
     }
 
+    /**
+     * Gets the alert reference.
+     *
+     * <p>This is a unique identifier for the type of alert raised. A scan rule may raise more that
+     * one type of alert but they should all have different alert references.
```
Comment (kingthorin) [inline zap/src/main/java/org/parosproxy/paros/core/scanner/Alert.java:949]:
```suggestion
     * one type of alert and they should all have different alert references.
```

File: zap/src/main/java/org/zaproxy/zap/extension/alert/AlertAPI.java [hunk 2]
```diff
@@ -474,6 +474,7 @@ private void processAlerts(
         Map<String, String> map = new HashMap<>();
         map.put("id", String.valueOf(alert.getAlertId()));
         map.put("pluginId", String.valueOf(alert.getPluginId()));
+        map.put("alertRef", alert.getAlertRef());
```
Comment (kingthorin) [inline zap/src/main/java/org/zaproxy/zap/extension/alert/AlertAPI.java:477]: Should we also update the core reporting? (Or separate issue/PR?)

Comment (psiinon) [reply zap/src/main/java/org/zaproxy/zap/extension/alert/AlertAPI.java:477]: Doh, forgot about those, will add to this PR...

Comment (psiinon) [reply zap/src/main/java/org/zaproxy/zap/extension/alert/AlertAPI.java:477]: Ha! No need - they now call the alert API endpoints so they dont need to change.

File: zap/src/main/java/org/parosproxy/paros/core/scanner/Alert.java (hunk 1 above)
Comment (psiinon) [reply zap/src/main/java/org/parosproxy/paros/core/scanner/Alert.java]: Done.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Comment by kingthorin** [comment]
You decided not to tackle core reports?

**Comment by psiinon** [comment]
You mean the HTML report? Thats really aimed at people and doesnt currently include the plugin id, so yes, wasnt planning on changing that...

**Review by kingthorin — APPROVED** [review]

**Pull request merged**